server/blob/blob-objectstorage/src/main/java/org/apache/james/blob/objectstorage/swift/SwiftTempAuthObjectStorage.java

```diff
@@ -0,0 +1,230 @@
+                userHeaderName = Optional.of(tmpAuthHeaderUser);
+                return this;
+            }
+
+            public Builder tempAuthHeaderPassName(PassHeaderName tmpAuthHeaderPass) {
+                passHeaderName = Optional.of(tmpAuthHeaderPass);
+                return this;
+            }
+
+            public Configuration build() {
+                Preconditions.checkState(endpoint != null);
+                Preconditions.checkState(tenantName != null);
+                Preconditions.checkState(userName != null);
+                Preconditions.checkState(credentials != null);
+                Identity identity = Identity.of(tenantName, userName);
+                return new Configuration(endpoint, identity, credentials, region,
+                    userHeaderName, passHeaderName);
+            }
+        }
+
+        private final URI endpoint;
+        private final Identity identity;
+        private final Optional<Region> region;
+        private final Credentials credentials;
+        private final Optional<UserHeaderName> userHeaderName;
+        private final Optional<PassHeaderName> passHeaderName;
+
+        private Configuration(URI endpoint,
+                              Identity identity,
+                              Credentials credentials,
+                              Optional<Region> region,
+                              Optional<UserHeaderName> userHeaderName,
+                              Optional<PassHeaderName> passHeaderName) {
+            this.endpoint = endpoint;
+            this.identity = identity;
+            this.region = region;
+            this.userHeaderName = userHeaderName;
+            this.passHeaderName = passHeaderName;
+            this.credentials = credentials;
+        }
+
+        public URI getEndpoint() {
+            return endpoint;
+        }
+
+        public Identity getIdentity() {
+            return identity;
+        }
+
+        public Credentials getCredentials() {
+            return credentials;
+        }
+
+        public Properties getOverrides() {
+            Properties properties = new Properties();
+            properties.setProperty(KeystoneProperties.CREDENTIAL_TYPE, "tempAuthCredentials");
+            userHeaderName.ifPresent(tmpAuthHeaderUser ->
+                properties.setProperty(TempAuthHeaders.TEMP_AUTH_HEADER_USER, tmpAuthHeaderUser.value())
+            );
+            passHeaderName.ifPresent(tmpAuthHeaderPass ->
+                properties.setProperty(TempAuthHeaders.TEMP_AUTH_HEADER_PASS, tmpAuthHeaderPass.value())
+            );
+            return properties;
+        }
+
+        public Optional<Region> getRegion() {
+            return region;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) {
+                return true;
+            }
+            if (o == null || getClass() != o.getClass()) {
+                return false;
```

Comment:
We tend to have a single **instanceOf** check in equals

The reference equality can be seen as an optimisation that may not be required and null check is handled by instanceOf.

Reply:
hmm, all the other value types in the package including ones which are already merged in master use this pattern (since it was generated by the intellij idea pattern) 
If you prefer me to change it everywhere I will create a commit on top to reflect that change.